notebooks/IMDB to Hotel Sentiment.ipynb

```diff
@@ -0,0 +1,725 @@
+    "    # The pipeline returns all of the label probabilities\n",
+    "    # We need to extract the largest\n",
+    "    max_score = 0\n",
+    "    label = None\n",
+    "    for l in label_probabilities:\n",
+    "        if l['score'] > max_score:\n",
+    "            max_score = l['score']\n",
+    "            label = l['label']\n",
+    "    return label\n",
+    "\n",
+    "y_pred = [get_label(x) for x in original_pipeline(test_data.sentence.to_list())]\n",
+    "\n",
+    "\n",
+    "test_data['my_y_pred'] = y_pred\n",
+    "assert np.array_equal(test_data['my_y_pred'], test_data['output'])\n",
+    "\n",
+    "display(test_data)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "385dbdff",
+   "metadata": {},
+   "source": [
+    "We can also evaluate our chosen metric, and check that the accuracy score matches that we expect from the summary at the top level of the AdaTest widget:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "cb2e1f62",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from datasets import load_metric\n",
+    "\n",
+    "metric_name = 'accuracy'\n",
+    "\n",
+    "metric = load_metric(metric_name)\n",
+    "\n",
+    "def label_to_int(l: str) -> int:\n",
+    "    # Use the mapping provided by the model\n",
+    "    return model.config.label2id[l]\n",
+    "\n",
+    "metric.compute(predictions=test_data['my_y_pred'].apply(label_to_int), references=test_data['label'].apply(label_to_int))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7d8348a9",
+   "metadata": {},
+   "source": [
+    "There is one final tweak to make to our data prior to finetuning the model: the Hugging Face `Trainer`s do not use the human-friendly labels, but the corresponding integer ids. So use the mapping provided by the model to convert the 'label' column:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ac67a09a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "test_data['label'] = test_data['label'].apply(label_to_int)\n",
+    "print(test_data.dtypes)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "184a4ee2",
+   "metadata": {},
+   "source": [
+    "Now, we can split our dataset into training and test sets:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "24150831",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sklearn.model_selection import train_test_split"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8bab0273",
+   "metadata": {},
+   "source": [
+    "We stratify based on the 'topic' column, to ensure that we have samples from all of the various topics we have generated:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "20d7fbf6",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_df, test_df = train_test_split(test_data, stratify=test_data['topic'], test_size=0.3)"
+   ]
+  },
```

Comment:
Let's merge these cells to keep things a bit more compact.

Reply:
OK